### PR TITLE
event-index: Pass the user/device id pair when initializing the event index.

### PR DESCRIPTION
**Pull request by poljar**
This extends the base event index manager to pass the user id and device id when initializing the event index, this is needed so we store the passphrase under a per user/device specific key.

This is a breaking change.

The element web part is found at https://github.com/vector-im/element-web/pull/15455.
The element desktop part is found at https://github.com/vector-im/element-desktop/pull/147.

### CODE REVIEW

File: src/indexing/BaseEventIndexManager.ts [hunk 1]
```diff
@@ -105,10 +105,13 @@ export default abstract class BaseEventIndexManager {
...
      * Initialize the event index for the given user.
      *
+     * @param {string} user_id The event that should be added to the index.
+     * @param {string} device_id The profile of the event sender at the
+     *
      * @return {Promise} A promise that will resolve when the event index is
      * initialized.
      */
-    async initEventIndex(): Promise<void> {
+    async initEventIndex(user_id: string, device_id: string): Promise<void> {
```
Comment (t3chguy) [inline src/indexing/BaseEventIndexManager.ts:114]: camelCase please

File: src/indexing/EventIndexPeg.js [hunk 2]
```diff
@@ -70,9 +71,13 @@ class EventIndexPeg {
     async initEventIndex() {
         const index = new EventIndex();
         const indexManager = PlatformPeg.get().getEventIndexingManager();
+        const client = MatrixClientPeg.get();
+
+        const user_id = client.getUserId();
+        const device_id = client.getDeviceId();
```
Comment (t3chguy) [inline src/indexing/EventIndexPeg.js:77]: same as above

File: src/indexing/EventIndexPeg.js [hunk 3]
```diff
@@ -70,9 +71,13 @@ class EventIndexPeg {
...
 
         try {
-            await indexManager.initEventIndex();
+            await indexManager.initEventIndex(user_id, device_id);
```
Comment (t3chguy) [inline src/indexing/EventIndexPeg.js:80]: and here

File: src/indexing/EventIndexPeg.js [hunk 4]
```diff
@@ -83,7 +88,7 @@ class EventIndexPeg {
                 await indexManager.closeEventIndex();
                 await this.deleteEventIndex();
 
-                await indexManager.initEventIndex();
+                await indexManager.initEventIndex(user_id, device_id);
```
Comment (t3chguy) [inline src/indexing/EventIndexPeg.js:91]: and here

**Comment by poljar** [comment]
@stoically  will this mess up radicale-native?

**Comment by stoically** [comment]
Seems non-breaking since the seshat API didn't change and the RN javascript part will just ignore the additional arguments - thanks for the heads-up! @poljar

**Pull request merged**